### PR TITLE
Stabilise LMS and further stabilise DPM++ to eliminate residual noise in the result

### LINKED ISSUE #5630: LMS scheduler leaves a lot of noise leftover in the result image when used with SDXL Img2Img Pipeline
### Describe the bug

When using the LMS scheduler with SDXL Img2Img pipeline, there is a lot of noise leftover in the image especially when `strength` is closer to `0`. In other words, when the total number of performed steps is "low" (e.g. `num_inference_steps=50` and `strength=0.1`), the result images are unusably noisy.

### Reproduction

Here's some code that first does a prompt-to-image generation, and then an image-to-image from that result with `strength =0.1`. The image-to-image result looks like an intermediate latent. Note that the prompt-to-image result looks completely fine. This is reproducible with any input image - I just used a p2i gen because it felt easier to share here.

```python3
import torch
from diffusers import StableDiffusionXLPipeline, StableDiffusionXLImg2ImgPipeline
from typing import cast
from diffusers import LMSDiscreteScheduler

sdxl_model = cast(StableDiffusionXLPipeline, StableDiffusionXLPipeline.from_pretrained(
    'stabilityai/stable-diffusion-xl-base-1.0',
    torch_dtype=torch.float16,
    use_safetensors=True,
    variant="fp16",
    revision="76d28af79639c28a79fa5c6c6468febd3490a37e",
)).to('cuda')
sdxl_img2img_model = cast(StableDiffusionXLImg2ImgPipeline, StableDiffusionXLImg2ImgPipeline.from_pretrained(
    'stabilityai/stable-diffusion-xl-base-1.0',
    torch_dtype=torch.float16,
    use_safetensors=True,
    variant="fp16",
    revision="76d28af79639c28a79fa5c6c6468febd3490a37e",
)).to('cuda')

common_config = {'beta_start': 0.00085, 'beta_end': 0.012, 'beta_schedule': 'scaled_linear'}
scheduler = LMSDiscreteScheduler(**common_config)
sdxl_model.scheduler = scheduler
sdxl_img2img_model.scheduler = scheduler

sdxl_model.watermark = None
generator = torch.Generator(device='cuda')
generator.manual_seed(12345)

params = {
    'prompt': ['evening sunset scenery blue sky nature, glass bottle with a galaxy in it'],
    'negative_prompt': ['text, watermark'],
    "negative_prompt": [''],
    "num_inference_steps": 50,
    "guidance_scale": 7,
    "width": 1024,
    "height": 1024
}
sdxl_res = sdxl_model(**params, generator=generator, output_type='pil')
sdxl_img = sdxl_res.images[0]
display(sdxl_img)

img2img_params = {
    'prompt': ['evening sunset scenery blue sky nature, glass bottle with a galaxy in it'],
    'negative_prompt': ['text, watermark'],
    "negative_prompt": [''],
    "num_inference_steps": 50,
    "guidance_scale": 7,
    "image": sdxl_img,
    "strength": 0.1
}

sdxl_img2img_res = sdxl_img2img_model(**img2img_params, generator=generator, output_type='pil')

display(sdxl_img2img_res.images[0])
```

Image-to-Image Result: 
image

### Logs

_No response_

### System Info

- `diffusers` version: 0.21.4
- Platform: Linux-5.4.0-163-generic-x86_64-with-glibc2.31
- Python version: 3.11.5
- PyTorch version (GPU?): 2.1.0+cu121 (True)
- Huggingface_hub version: 0.17.1
- Transformers version: 4.34.0
- Accelerate version: 0.22.0
- xFormers version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

### Who can help?

@yiyixuxu @patrickvonplaten

### PR DESCRIPTION
**Pull request by nhnt11**
# What does this PR do?

This PR makes two changes:
1. For the LMS scheduler, introduces a parameter that forces `order=1` for the last 15 steps. This makes LMS usable with Image to Image with strength closer to 0 (e.g. 0.1). This solves https://github.com/huggingface/diffusers/issues/5630 (more example results in that issue's thread)
2. For the DPMSolverMultistep scheduler, we force the last sigma to be `0.0` if `euler_at_final` is true. This prevents residual noise in the result image. This solves https://github.com/huggingface/diffusers/issues/5689 (more example results in that issue's thread)

Fixes #5630, #5689

### CODE REVIEW

**Comment by nhnt11** [comment]
I am still working on cleaning this up a bit and updating documentation for the LMS config param 🙏

**Comment by patrickvonplaten** [comment]
@yiyixuxu could you take a look here?

**Comment by patrickvonplaten** [comment]
Gentle ping here @yiyixuxu

**Comment by yiyixuxu** [comment]
hi sorry I didn't have time to look into this sooner. We got a couple more bug reports on DPM scheduler with `use_karras_sigmas`, so I looked into it more. I think there was an issue in our implementation with `use_karras_sigmas`, and you are absolutely correct that the last sigma should be `0` (that's how it is done in k-diffusions)

I'm working on fixing it with PR here https://github.com/huggingface/diffusers/pull/6477
